### PR TITLE
Report Exceptions

### PR DESCRIPTION
**Pull request by evolvedmicrobe**
* Keeps track of unexpected exceptions
* Outputs exception results and other results.

### CODE REVIEW

File: src/main/ccs.cpp [hunk 1]
```diff
@@ -273,6 +273,13 @@ void WriteResultsReport(ostream& report, const Results& counts)
 
     report << "Failed -- CCS below minimum predicted accuracy," << counts.PoorQuality << ","
            << 100.0 * counts.PoorQuality / total << '%' << endl;
+    
+    report << "Failed -- Exception thrown during processing," << counts.ExceptionThrown << ","
+           << 100.0 * counts.ExceptionThrown / total << '%' << endl;
+    
+    report << "Failed -- Miscellaneous reasons" << counts.Other << ","
+           << 100.0 * counts.Other / total << '%' << endl;    
+
```
Comment (nlhepler) [inline src/main/ccs.cpp]: Do we need Other anymore? It looks from here like ExceptionThrown is covering all our remaining bases now.

Comment (evolvedmicrobe) [inline src/main/ccs.cpp]: Yeah, I just searched for it and didn't find it anywhere so am assuming we don't use it.  I left it in for future reasons, but suppose we can just take it out.

File: include/pacbio/ccs/Consensus.h [hunk 2]
```diff
@@ -518,9 +519,18 @@ ResultType<TResult> Consensus(std::unique_ptr<std::vector<TChunk>>& chunksRef,
             results.emplace_back(TResult{chunk.Id, std::string(ai), QVsToASCII(qvs), nPasses,
                                          predAcc, zAvg, zScores, statusCounts, nTested, nApplied,
                                          chunk.SignalToNoise, timer.ElapsedMilliseconds(), chunk.Barcodes});
-        } catch (...) {
-            results.Other += 1;
+        }
+        catch (const std::exception &exc) {
+            results.ExceptionThrown += 1;
             PBLOG_ERROR << "Skipping " << chunk.Id << ", caught exception during processing";
+            PBLOG_ERROR << "Exception was: " << exc.what();
```
Comment (nlhepler) [inline include/pacbio/ccs/Consensus.h]: I'm not sure we want this as two independent events, could you please make it a single two-line event?

File: include/pacbio/ccs/Consensus.h [hunk 3]
```diff
@@ -179,8 +179,8 @@ class ResultType : public std::vector<TConsensus>
     size_t TooManyUnusable;
     size_t NonConvergent;
     size_t PoorQuality;
-    size_t Other;
-
+    size_t ExceptionThrown;
+
```
Comment (nlhepler) [inline include/pacbio/ccs/Consensus.h]: so, a few niggling things: dangling whitespace

File: include/pacbio/ccs/Consensus.h (hunk 2 above)
Comment (evolvedmicrobe) [inline include/pacbio/ccs/Consensus.h]: yeah, I am actually not sure how you setup the logging.  Is 1 line guaranteed to be thread safe?  How are 2 simulatenous << processed?

Comment (nlhepler) [inline include/pacbio/ccs/Consensus.h]: All are guaranteed thread-safe. mutex-guarded moves of unique_ptrs of ostringstreams

**Comment by nlhepler** [comment]
Sweetness

**Pull request merged**